pkg/audit/stats_reporter.go

```diff
@@ -87,16 +126,24 @@ func (r *reporter) observeRunEnd(_ context.Context, o metric.Float64Observer) er
 }
 
 // newStatsReporter creates a reporter for audit metrics.
-func newStatsReporter() (*reporter, error) {
-	r := &reporter{}
+func newStatsReporter(enableConstraintLabelMetrics bool) (*reporter, error) {
+	r := &reporter{enableConstraintLabelMetrics: enableConstraintLabelMetrics}
 	var err error
 	meter := otel.GetMeterProvider().Meter("gatekeeper")
 
-	_, err = meter.Int64ObservableGauge(
-		violationsMetricName,
-		metric.WithDescription("Total number of audited violations"),
-		metric.WithInt64Callback(r.observeTotalViolations),
-	)
+	if enableConstraintLabelMetrics {
```

Comment:
This will result in reporting any one of the metrics at a time depending on the flag value, either violations per constraint or total violations. Rather this should be another metrics `gatekeeper_violation_per_constraint` and it should be only reported when the flag is true. `gatekeeper_violations` should be reported as is. You will need to use another metrics name const for this.

Reply:
Hi Jaydip, thanks for reviewing - I thought having the extra dimensionality makes sense here as it will be easy for people to migrate existing tooling if they want this extra detail. If someone enables the flag, emitting `gatekeeper_violation_per_constraint` and `gatekeeper_violations` causes waste, as they can still query `gatekeeper_violations` with `sum without (constraint) (gatekeeper_violation_per_constraint)`. I still think adding the (optional) label for `constraint` to the metric with the same name makes sense, but am happy to change the name as you suggested.

Reply:
Will leave existing behaviour in place as suggested and just add `gatekeeper_violation_per_constraint`